File: app/src/main/java/com/example/coronatracker/Room/viewModel.java
package com.example.coronatracker.Room;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class viewModel extends AndroidViewModel {
    private repo repository;
    private LiveData<List<indiaStateModel>> liveListData;
    public viewModel(@NonNull Application application) {
        super(application);
        this.repository=new repo(application);
        this.liveListData=this.repository.getOfflineData();
    }
    public void insert(indiaStateModel model){
        this.repository.insert(model);
    }
    public LiveData<List<indiaStateModel>> getOfflineData(){
        return this.liveListData;
    }
    public void deleteAll(){
        this.repository.deleteAll();
    }
}

File: app/src/main/java/com/example/coronatracker/room/ViewModel.kt
package com.example.coronatracker.room


import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData


class ViewModel(application: Application) : AndroidViewModel(application) {
    private val repository: Repo = Repo(application)
    val offlineData: LiveData<List<indiaStateModel?>?>? = repository.offlineData

    suspend fun insert(model: indiaStateModel?) = repository.insert(model)

    suspend fun getOfflineDataB() : List<indiaStateModel?> = repository.getOfflineDataB()

    suspend fun deleteAll() = repository.deleteAll()

    suspend fun insertState(state: StateRecent) = repository.insertState(state)

    suspend fun insertCountry(country: CountryRecent) = repository.insertCountry(country)

    fun getStateRecent() : LiveData<List<StateRecent>> =repository.getRecentState()

    fun getCountryRecent(): LiveData<List<CountryRecent>> =repository.getRecentCountry()
}